src/tools/rados/rados.cc

```diff
@@ -2034,7 +2035,8 @@ static int rados_tool_common(const std::map < std::string, std::string > &opts,
   i = opts.find("format");
   if (i != opts.end()) {
     const char *format = i->second.c_str();
-    formatter = Formatter::create(format);
+    _formatter.reset(Formatter::create(format));
+    formatter = _formatter.get();
```

Comment:
we assign to `formatter` at multiple places, like
```c++
    if (!formatter) {
      formatter = new JSONFormatter(pretty_format);
    }
```
so this change leaks.

instead, i'd suggest put just use the unique_ptr when referencing the Formatter, instead of using the raw pointer.

and update the places like above to something like
```c++
    if (!formatter) {
      formatter = std::make_unique<JSONFormatter>(pretty_format);
    }
```

Reply:
those places revised already, and some functions take raw pointer as input, so we have to leave the raw pointer, otherwise not compile